server/src/grpc/meta_event_service/shard_operation.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Implementation of operations on shards.
+
+// TODO: Currently, only a specific operation (close wal region) is implemented,
+// and it is expected to encapsulate more operations on **Shard** in the future.
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use common_types::table::ShardId;
+use common_util::error::{BoxError, GenericResult};
+use wal::manager::WalManagerRef;
+
+#[async_trait]
+pub trait WalRegionCloser: std::fmt::Debug + Send + Sync {
```

Comment:
Why not define a `ShardOperator`?
And we place `open_tables_on_shard` and `close_tables_on_shard` into it?

Reply:
I think if all actions about opening and closing should be placed together.
Otherwise, it's easy to forget to modify some of them in later?

Reply:
Actually, `ShardOperator` needs more refactoring, which includes not only operation about wal region but also some other operations about the engine. And I will do this in the following PRs.

Reply:
Ok.